src/hooks/use-fetch.ts

```diff
@@ -0,0 +1,19 @@
+import { usePromise } from "@nebula.js/stardust";
+
+export default function useFetch<T>(
+  fetch: () => Promise<T>,
+  deps: unknown[]
+): [result: T | undefined, isLoading: boolean, error: Error | undefined] {
+  const [response, error] = usePromise(async () => {
+    const result = await fetch();
+
+    return {
+      result,
+      deps,
+    };
+  }, deps);
+
+  const isLoading = response === undefined || response.deps.some((dep, idx) => dep !== deps[idx]);
```

Comment:
I think loading state could be incorrect because of: 
1. if the response is void on some rpc/api call we will always have isLoading as true?
2. object coercions on deps array (some crazy deps scenario like below may cause is loading to be true?)

<img width="505" alt="Screenshot 2023-08-22 at 10 15 49" src="https://github.com/qlik-oss/sn-pivot-table/assets/29652890/e3463308-3436-48ac-8c94-f47b1b52aade">


Maybe a solution would be to wrap await part into a try catch and use a state for loading. Then in finally of that try...catch call set that flag to false?

Reply:
1. Good catch! I think a fix is to simple check if there is an error, than loading is "done". Using state will trigger an extra render when loading is finsished so want to avoid that.
2. I'll look into it